powertools-sqs/src/test/java/software/amazon/lambda/powertools/sqs/internal/SqsMessageBatchProcessorAspectTest.java

```diff
@@ -0,0 +1,115 @@
+package software.amazon.lambda.powertools.sqs.internal;
+
+import java.io.IOException;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.SQSEvent;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import software.amazon.awssdk.services.sqs.SqsClient;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageBatchRequest;
+import software.amazon.awssdk.services.sqs.model.GetQueueUrlRequest;
+import software.amazon.awssdk.services.sqs.model.GetQueueUrlResponse;
+import software.amazon.lambda.powertools.sqs.SQSBatchProcessingException;
+import software.amazon.lambda.powertools.sqs.handlers.LambdaHandlerApiGateway;
+import software.amazon.lambda.powertools.sqs.handlers.PartialBatchFailureSuppressedHandler;
+import software.amazon.lambda.powertools.sqs.handlers.PartialBatchPartialFailureHandler;
+import software.amazon.lambda.powertools.sqs.handlers.PartialBatchSuccessHandler;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+import static software.amazon.lambda.powertools.sqs.PowertoolsSqs.defaultSqsClient;
+
+public class SqsMessageBatchProcessorAspectTest {
+    public static final SqsClient sqsClient = mock(SqsClient.class);
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    private SQSEvent event;
+    private RequestHandler<SQSEvent, String> requestHandler;
+
+    private Context context = mock(Context.class);
+
+    @BeforeEach
+    void setUp() throws IOException {
+        defaultSqsClient(sqsClient);
+        reset(sqsClient);
+        setupContext();
+        event = MAPPER.readValue(this.getClass().getResource("/sampleSqsBatchEvent.json"), SQSEvent.class);
+
+        when(sqsClient.getQueueUrl(any(GetQueueUrlRequest.class))).thenReturn(GetQueueUrlResponse.builder()
+                .queueUrl("test")
+                .build());
+
+        requestHandler = new PartialBatchSuccessHandler();
+    }
+
+    @Test
+    void shouldBatchProcessAllMessageSuccessfullyAndNotDeleteFromSQS() {
+        requestHandler.handleRequest(event, context);
+
+        verify(sqsClient, times(2)).listQueues();
```

Comment:
I had to look up why `listQueues();` was being called. I can see it's just for testing but maybe there is a less confusing way to do it?

Reply:
Use Random as a mock totally unrelated to make it less confusing.